`AbortSignal.timeout`  not working with typescript

### What version of Bun is running?

1.2.7+5c0fa6dc2

### What platform is your computer?

Linux 5.15.167.4-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

1. Create a new project with `bun init`
2. Add `AbortSignal.timeout(1);` to index.ts
3. Run `bunx tsc`

### What is the expected behavior?

Should compile with `tsc`

### What do you see instead?

error TS2339: Property 'timeout' does not exist on type '{ new (): AbortSignal; prototype: AbortSignal; }

### Additional information

Using `@types/bun@1.2.7`

--- a/packages/bun-types/globals.d.ts
+++ b/packages/bun-types/globals.d.ts
@@ -1384,7 +1384,7 @@ interface Uint8ArrayConstructor {
 interface BroadcastChannel {}
 declare var BroadcastChannel: Bun.__internal.UseLibDomIfAvailable<
   "BroadcastChannel",
-  import("node:worker_threads").BroadcastChannel
+  typeof import("node:worker_threads").BroadcastChannel
 >;
 
 declare var URL: Bun.__internal.UseLibDomIfAvailable<
@@ -1434,6 +1434,24 @@ declare var AbortSignal: Bun.__internal.UseLibDomIfAvailable<
   {
     prototype: AbortSignal;
     new (): AbortSignal;
+    /**
+     * Create an AbortSignal that will be aborted after a timeout
+     * @param ms The timeout in milliseconds
+     * @returns An AbortSignal that will be aborted after the timeout
+     */
+    timeout(ms: number): AbortSignal;
+    /**
+     * Create an immediately-aborted AbortSignal
+     * @param reason The reason for the abort
+     * @returns An AbortSignal that is already aborted
+     */
+    abort(reason?: any): AbortSignal;
+    /**
+     * Create an AbortSignal that will be aborted if any of the signals are aborted
+     * @param signals The signals to combine
+     * @returns An AbortSignal that will be aborted if any of the signals are aborted
+     */
+    any(signals: AbortSignal[]): AbortSignal;
   }
 >;
 

--- a/test/integration/bun-types/fixture/index.ts
+++ b/test/integration/bun-types/fixture/index.ts
@@ -11,6 +11,19 @@ const options: Bun.TLSOptions = {
 
 process.assert;
 
+const channel = new BroadcastChannel("my-channel");
+const message = { hello: "world" };
+
+channel.onmessage = event => {
+  console.log(event);
+};
+
+channel.postMessage(message);
+
+const error = new Error("hello world");
+const clone = structuredClone(error);
+console.log(clone.message); // "hello world"
+
 new SubtleCrypto();
 declare const mySubtleCrypto: SubtleCrypto;
 
@@ -78,6 +91,9 @@ new WritableStream();
 new TransformStream();
 new AbortSignal();
 new AbortController();
+AbortSignal.timeout(200);
+AbortSignal.any([new AbortSignal()]);
+AbortSignal.abort(200);
 
 new TextDecoder();
 new TextEncoder();

--- a/test/integration/bun-types/fixture/package.json
+++ b/test/integration/bun-types/fixture/package.json
@@ -6,7 +6,6 @@
   },
   "type": "module",
   "dependencies": {
-    "@types/bun": "types-bun-1.2.6.tgz",
     "typescript": "latest"
   }
 }

